netman/adapters/switches/juniper/base.py

```diff
@@ -522,15 +526,14 @@ def add_interface_to_bond(self, interface, bond_id):
         self._push_interface_update(interface, update)
 
     def remove_interface_from_bond(self, interface):
-        config = self.query(all_interfaces)
-        node = self.get_interface_config(interface, config)
-
-        if len(node.xpath('ether-options/ieee-802.3ad')) is 0:
-            raise InterfaceNotInBond
+        try:
+            update = Update()
+            update.add_interface(free_from_bond_operation(interface))
+            self._push(update)
```

Comment:
The error we get for : no interface is vs no 802 is set is the same error so need another call to discern